.github/workflows/maven.yaml

```diff
@@ -0,0 +1,133 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This workflow will build a Java project with Maven
+# See also:
+#   https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: QA
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+permissions:
+  contents: read
+
+jobs:
+  # fast build to populate the local maven repository cache
+  fastbuild:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        distribution: adopt
+        java-version: 11
+    - name: Cache local maven repository
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.m2/repository/
+          !~/.m2/repository/org/apache/accumulo
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Show the first log message
+      run: git log -n1
+    - name: Check for unapproved characters
+      run: src/build/ci/find-unapproved-chars.sh
+    - name: Check for unapproved JUnit API usage
+      run: src/build/ci/find-unapproved-junit.sh
+    - name: Build with Maven (Fast Build)
+      timeout-minutes: 20
+      run: mvn -B -V -e -ntp "-Dstyle.color=always" clean package dependency:resolve -DskipTests -DskipFormat -DverifyFormat
+      env:
+        MAVEN_OPTS: -Djansi.force=true
+  # more complete builds with tests
+  mvn:
+    needs: fastbuild
+    strategy:
+      matrix:
+        profile:
+          - {name: 'unit-tests',    javaver: 11, args: 'verify -PskipQA -DskipTests=false'}
+          - {name: 'qa-checks',     javaver: 11, args: 'verify javadoc:jar -Psec-bugs -DskipTests -Dspotbugs.timeout=3600000'}
+          - {name: 'errorprone',    javaver: 11, args: 'verify -Perrorprone,skipQA'}
+          - {name: 'jdk17',         javaver: 17, args: 'verify -DskipITs'}
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK ${{ matrix.profile.javaver }}
+      uses: actions/setup-java@v3
+      with:
+        distribution: adopt
+        java-version: ${{ matrix.profile.javaver }}
+    - name: Cache local maven repository
+      uses: actions/cache@v3
+      witi:
+        path: |
+          ~/.m2/repository/
+          !~/.m2/repository/org/apache/accumulo
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Override DNS to fix IP address for hostname
```

Comment:
I wonder if this was added so that the mini test could run, may be able to drop this section

Reply:
No idea, I'm going to leave this.

Reply:
After we get some initial gh action builds running it will be easy to experiment with removing it in a PR